apis/builder/validators.yaml

```
@@ -0,0 +1,37 @@
+post:
+  operationId: "registerValidator"
+  summary: Register or update a validator's block building preferences.
+  description: |
+    Registers a validator's preferred fee recipient and gas limit.
+
+    Builders should verify that `pubkey` corresponds to an active or pending
+    validator, and that `signature` is valid under `pubkey`. Otherwise, builder
+    must error. Requests with `timestamp` less than or equal to the previous
+    announcement must error. Requests with `timestamp` more than 10 seconds in
+    the future must error.
+  tags:
+    - Builder
+  requestBody:
+    description: An array of signed validator block building preferences.
```

It's unclear to me the value of making this an array rather than a singleton especially in light of potentially sending this to gossip. In the event that this registration is gossiped, it would certainly be as singletons.

That said, mev-boost could split into singletons and gossip.

---

The structure here just mirrors the signature in https://github.com/ethereum/beacon-APIs/pull/209

and the rationale is to allow nodes with many validators to make one request, rather than make many requests one-by-one

---

we can also more naturally get fine-grained error reporting if we move to the individual style interface

right now you can return an error from this endpoint but there doesn't seem to be a way to specify which registration in the batch failed (unless identifying info, like pubkey, is in the error message)

---

remove in 565d1db